test/python/circuit/test_isometry.py

```diff
@@ -133,6 +133,24 @@ def test_isometry_inverse(self, iso):
         result = Operator(qc)
         np.testing.assert_array_almost_equal(result.data, np.identity(result.dim[0]))
 
+    @data(
+        np.eye(2, 2),
+        random_unitary(2, seed=297102).data,
+        np.eye(4, 4),
+        random_unitary(4, seed=123642).data,
+        random_unitary(8, seed=568288).data,
+    )
+    def test_isometry_repeat(self, iso):
+        """Tests for the repeat of isometries from n to n qubits"""
+        if len(iso.shape) == 1:
+            iso = iso.reshape((len(iso), 1))
```

Comment:
the unitaries you are defining in the data all have shape != 1, don't they? in that case we can skip these lines:

```suggestion
```